src/expr/src/relation/mod.rs

```diff
@@ -2663,6 +2668,8 @@ pub struct RowSetFinishing {
     pub offset: usize,
     /// Include only given columns.
     pub project: Vec<usize>,
+    /// User-supplied hint: how many rows will have the same group key.
+    pub expected_group_size: Option<u64>,
```

Comment:
I'm a bit surprised to see this field added here. I understand `RowSetFinishing` to describe operations that should modify dataflow results. `expected_group_size` informs rendering decisions so it doesn't seem to fit here. Is it only added to return the group size from `plan_query`?

Reply:
Originally, plumbing `expected_group_size` through `plan_query_inner` directly seemed quite hideous, but I just tried it again and it's not bad, so I'll do that instead.  I expect that fourth tuple element will eventually need to become a catch-all options structure but I'll avoid doing that presently.